Library/Homebrew/tab.rb

```diff
@@ -35,30 +35,39 @@ def self.from_file(path)
     from_file_content(File.read(path), path)
   end
 
-  def self.from_file_content(content, path)
-    attributes = Utils::JSON.load(content)
-    attributes["tabfile"] = path
-    attributes["source"] ||= {}
-
-    tapped_from = attributes["tapped_from"]
-    unless tapped_from.nil? || tapped_from == "path or URL"
-      attributes["source"]["tap"] = attributes.delete("tapped_from")
-    end
+  def self.write_file_content(content, path)
+    from_file_content(content, path, :refresh_cache => :true)
+    path.atomic_write(content)
```

Comment:
This doesn't read the old tab file as `self.from_file_content` is passed the new path/content and reads from the passed content rather than from disk.